modules/mono/glue/GodotSharp/GodotSharp/Core/Projection.cs

```diff
@@ -0,0 +1,817 @@
+        {
+            x = new Vector4(transform.basis.Row0.x, transform.basis.Row1.x, transform.basis.Row2.x, 0);
+            y = new Vector4(transform.basis.Row0.y, transform.basis.Row1.y, transform.basis.Row2.y, 0);
+            z = new Vector4(transform.basis.Row0.z, transform.basis.Row1.z, transform.basis.Row2.z, 0);
+            w = new Vector4(transform.origin.x, transform.origin.y, transform.origin.z, 1);
+        }
+
+        /// <summary>
+        /// Constructs a new <see cref="Transform3D"/> from the <see cref="Projection"/>.
+        /// </summary>
+        /// <param name="proj">The <see cref="Projection"/>.</param>
+        public static explicit operator Transform3D(Projection proj)
+        {
+            return new Transform3D(
+                new Basis(
+                    new Vector3(proj.x.x, proj.x.y, proj.x.z),
+                    new Vector3(proj.y.x, proj.y.y, proj.y.z),
+                    new Vector3(proj.z.x, proj.z.y, proj.z.z)
+                ),
+                new Vector3(proj.w.x, proj.w.y, proj.w.z)
+            );
+        }
+
+        public static Projection CreateDepthCorrection(bool flipY)
+        {
+            return new Projection(
+                new Vector4(1, 0, 0, 0),
+                new Vector4(0, flipY ? -1 : 1, 0, 0),
+                new Vector4(0, 0, (real_t)0.5, 0),
+                new Vector4(0, 0, (real_t)0.5, 1)
+            );
+        }
+
+        public static Projection CreateFitAabb(AABB aabb)
+        {
+            Vector3 min = aabb.Position;
+            Vector3 max = aabb.Position + aabb.Size;
+
+            return new Projection(
+                new Vector4(2 / (max.x - min.x), 0, 0, 0),
+                new Vector4(0, 2 / (max.y - min.y), 0, 0),
+                new Vector4(0, 0, 2 / (max.z - min.z), 0),
+                new Vector4(-(max.x + min.x) / (max.x - min.x), -(max.y + min.y) / (max.y - min.y), -(max.z + min.z) / (max.z - min.z), 1)
+            );
+        }
+
+        public static Projection CreateForHmd(int eye, real_t aspect, real_t intraocularDist, real_t displayWidth, real_t displayToLens, real_t oversample, real_t zNear, real_t zFar)
+        {
+            real_t f1 = (intraocularDist * (real_t)0.5) / displayToLens;
+            real_t f2 = ((displayWidth - intraocularDist) * (real_t)0.5) / displayToLens;
+            real_t f3 = (displayWidth / (real_t)4.0) / displayToLens;
+
+            real_t add = ((f1 + f2) * (oversample - (real_t)1.0)) / (real_t)2.0;
+            f1 += add;
+            f2 += add;
+            f3 *= oversample;
+
+            f3 /= aspect;
+
+            switch (eye)
+            {
+                case 1:
+                    return CreateFrustum(-f2 * zNear, f1 * zNear, -f3 * zNear, f3 * zNear, zNear, zFar);
+                case 2:
+                    return CreateFrustum(-f1 * zNear, f2 * zNear, -f3 * zNear, f3 * zNear, zNear, zFar);
+                default:
+                    GD.PushError("eye is not 1 or 2");
```

Comment:
Since the C++ equivalent doesn't push an error here, I don't think C# should either.

https://github.com/godotengine/godot/blob/8cb51a642e7c34a73e7ed877ab46c041521205e5/core/math/projection.cpp#L339-L340